src/components/CheckForm/FormLayout/FormLayout.tsx

```diff
@@ -1,42 +1,117 @@
+
+import { FormSidebar } from './FormSidebar';
 
 type FormLayoutProps = {
   children: ReactNode;
 };
 
-export const FormLayout = ({ children }: FormLayoutProps) => {
+export const FormLayout = ({ children, formActions }: FormLayoutProps & CheckFormTypeLayoutProps) => {
   let index = -1;
+  const [activeIndex, setActiveIndex] = useState(0);
+  const sectionHeaders: Array<{ label: string; hasErrors: boolean; required: boolean }> = [];
+  const { formState, trigger } = useFormContext<CheckFormValues>();
+  const styles = useStyles2(getStyles);
 
-  return Children.map(children, (child) => {
+  let sectionCount = 0;
+  Children.forEach(children, (child) => {
+    if (!isValidElement(child)) {
+      return;
+    }
+    if (child?.type === FormSection) {
+      sectionCount++;
+    }
+  });
+  const sections = Children.map(children, (child) => {
     if (!isValidElement(child)) {
       return null;
     }
 
     if (child.type === FormSection) {
       index++;
 
-      return <FormSectionInternal {...child.props} index={index} />;
+      const errors = checkForErrors(formState.errors, child.props.fields);
+      sectionHeaders.push({ label: child.props.label, hasErrors: errors.length > 0, required: child.props.required });
+      return (
+        <FormSectionInternal
+          {...child.props}
+          index={index}
+          active={activeIndex === index}
+          onNextClick={
+            index !== sectionCount - 1
+              ? () => {
+                  trigger(child.props.fields).then((valid) => {
+                    if (valid) {
+                      setActiveIndex(activeIndex + 1);
+                    }
+                  });
+                }
+              : undefined
+          }
+        />
+      );
     }
 
     return child;
   });
+
+  return (
+    <div className={css({ display: 'flex', flexDirection: 'row', height: '100%' })}>
+      <FormSidebar sections={sectionHeaders} onSectionSelect={setActiveIndex} activeIndex={activeIndex} />
+      <div
+        className={css({
+          display: 'flex',
+          flexDirection: 'column',
+          flexGrow: '1',
+          justifyContent: 'space-between',
+        })}
+      >
+        <div className={css({ paddingLeft: '24px' })}>{sections}</div>
+        <div>
+          <hr className={css({ width: '100%' })} />
+          <div className={css({ display: 'flex', justifyContent: 'space-between', bottom: '0' })}>
+            <div className={styles.buttonGroup}>
+              {activeIndex !== 0 && (
+                <Button onClick={() => setActiveIndex(activeIndex - 1)} icon="arrow-left" variant="secondary">
+                  {sectionHeaders[activeIndex - 1].label}
+                </Button>
+              )}
+              {activeIndex !== sectionHeaders.length - 1 && (
+                <Button
+                  onClick={() => {
+                    trigger(sections?.[activeIndex].props.fields).then((valid) => {
+                      if (valid) {
```

Comment:
I don't think we should block advancing in the form if the section isn't valid. We don't block it when they are using the sidebar and some people might just want to explore the form options before they fill it all in.